Sources/NIOPosix/SelectableEventLoop.swift

```diff
@@ -31,6 +31,30 @@ internal func withAutoReleasePool<T>(_ execute: () throws -> T) rethrows -> T {
 #endif
 }
 
+/// Information about an EventLoop tick
+public struct NIOEventLoopTickInfo: Sendable, Hashable {
+    /// The eventloop which ticked
+    public var eventLoopID: ObjectIdentifier
+    /// The number of tasks which were executed in this tick
+    public var numberOfTasks: Int
+    /// The time at which the tick began
+    public var startTime: NIODeadline
+
+    internal init(eventLoopID: ObjectIdentifier, numberOfTasks: Int, startTime: NIODeadline) {
+        self.eventLoopID = eventLoopID
+        self.numberOfTasks = numberOfTasks
+        self.startTime = startTime
+    }
+}
+
+/// Implement this delegate to receive information about the EventLoop, such as each tick
+public protocol NIOEventLoopMetricsDelegate: Sendable {
+    /// Called after a tick has run
+    /// This function is called after every tick - avoid long-running tasks here
```

Comment:
Let's be really clear in this API, add a big warning block that says something like: "This function is called after every event loop tick and on the event loop thread. Any non-trivial work in this function will block the event loop and cause latency increases and performance degradation."